### DISCUSSION TITLE
[Forwards] Source and Destination

### DISCUSSION THREAD

**Discussion by surendra841** [aahnik/tgcf#83] [Q&A] [answered]
I understand it accept only integers. 

Are these only limited to Channels/groups? If not, please continue

I used  userID's (from @userinfobot) on source and destination.

It doesn't show any errors in live mode and the messages are not forwarded as well.

In the past mode, I get following error

> _telethon.errors.rpcerrorlist.ChatIdInvalidError: Invalid object ID for a chat. Make sure to pass the right types, for instance making sure that the request is designe
d for chats (not channels/megagroups) or otherwise look for a different one more suited\nAn example working with a megagroup and AddChatUserRequest, it will fail beca
use megagroups are channels. Use InviteToChannelRequest instead (caused by GetHistoryRequest)
_

I'm assuming its checking the userID's that I have input as chatID's?

Can anyone please tell help, thanks.

**Comment by aahnik** [comment, marked as answer]
Its no longer required to use chat ids. (if you want you may use)

You may directly use **username, or phone no, or link.**

tgcf works for normal chat, group, channels, and also for other bots.(when you are logged in with user account)

you must not use user ids . and chat ids start with `-100`
